elastic/kibana
# [Security Solution] Improve alert UX for criticality enrichments

## Summary

### Assign user-friendly column labels to enrichment fields for improved readability.
* Add a `displayAsText` prop for Entity Analytics fields on the Alerts table.
  *  host.asset.criticality -> Host Criticality
  * user.asset.criticality -> User Criticality
  * host.risk.calculated.level -> Host Risk Level
  * user.risk.calculated.level -> User Risk Level
* Add migration that renames `host.risk.calculated._evel` and `user.risk.calculated_level` on local storage

### Render a custom component for the Host and User criticality inside the Alerts table cell.
![Screenshot 2024-02-02 at 13 03 21](https://github.com/elastic/kibana/assets/1490444/91db2dec-6fe5-4a09-9a8e-b55be4ef927d)

### How to test it?
* Clone and install https://github.com/elastic/security-documents-generator
* Execute `npm start entity-store` on an empty kibana instance
* Increase the created rule `Additional look-back time` and restart the rule (that will generate more alerts)
* Open the alerts page (make sure your local storage is empty otherwise, you won't see the new column names)

### Know issues with the current implementation
* The new columns might not be displayed to users after an upgrade
* We couldn't rename `calculated_score_norm` and `calculated_score_norm` because they are not shown by default



### Context

#### Known Alert columns issues:
_The columns are stored on local storage when the alerts page loads_

* After users open the Alerts page once, they will always see the stored columns 
* We need to create a migration when adding or updating a default column.
* The risk score columns are displayed for users that have the risk engine disabled.

#### Risk fields friendly names limitations
* We can only rename columns that are displayed by default.
* If the user has loaded the page, he will only see the new column name if we write a migration to rename the columns inside local storage.

This [issue](https://github.com/elastic/kibana/issues/176125) tracks the @elastic/response-ops team's planned improvements to column names and visibility.



### Checklist

Delete any items that are not applicable to this PR.
